backup/tar.go

```diff
@@ -119,8 +122,19 @@ func extractTar(tarball []byte, chown bool, dirpath string, filesystem output.Fi
 				return errors.Wrapf(err, "error reading %s", header.Name)
 			}
 
-			if header.Name != filepath.Clean(header.Name) {
-				return fmt.Errorf("non-canonical file path in archive: %s", header.Name)
+			// To avoid path-traversal issues a la CVE-2021-43798, prepend a '/' to the front
+			// of the name before checking to see if it canonical, since by default filepath.Clean
+			// does not remove .. at the beginning of a path unless it is rooted.
```

Comment:
Would it be worth including a link to the CVE on NVD or Mitre?

Also, small typo: should be "it's canonical"

Could also link to the filepath.Clean docs if this behavior is documented.

Reply:
Added links.